osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -373,6 +350,15 @@ protected override void Update()
                 X = hyperDashTargetPosition;
                 SetHyperDashState();
             }
+
+            if (Dashing || HyperDashing)
+            {
+                double lastTrailTime = trails.LastDashTrail?.LifetimeStart ?? double.NegativeInfinity;
+                double generationInterval = HyperDashing ? 25 : 50;
+
+                if (Time.Current - lastTrailTime >= generationInterval)
```

Comment:
I guess this `lastTrailTime` trick sort of works (if there was no last trail, the left hand side of the comparison will be `Time.Current - (-inf) = +inf`), but really wouldn't it be simpler and clearer to just null-check it at this point?